File: backend-springboot/src/main/java/com/esprit/examen/entities/Reclamation.java
package com.esprit.examen.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Reclamation implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idReclamation;

    @ManyToOne(optional = false)
    private Operateur operateur;

    @ManyToOne(optional = false)
    private Commande commande;

    private String objet;

    @Column(length = 2000)
    private String description;

    @Enumerated(EnumType.STRING)
    private ReclamationStatus status = ReclamationStatus.OUVERT;

    private Instant createdAt = Instant.now();
    private Instant updatedAt = Instant.now();
}

File: frontend-flutter/lib/models/reclamation.dart
import 'commande.dart';
import 'operateur.dart';

class Reclamation {
  final int idReclamation;
  final Operateur operateur;
  final Commande commande;
  final String objet;
  final String description;
  final String status;
  final DateTime? createdAt;

  Reclamation({
    required this.idReclamation,
    required this.operateur,
    required this.commande,
    required this.objet,
    required this.description,
    required this.status,
    this.createdAt,
  });

  factory Reclamation.fromJson(Map<String, dynamic> json) {
    return Reclamation(
      idReclamation: (json['idReclamation'] ?? 0) as int,
      operateur: Operateur.fromJson((json['operateur'] ?? {}) as Map<String, dynamic>),
      commande: Commande.fromJson((json['commande'] ?? {}) as Map<String, dynamic>),
      objet: (json['objet'] ?? '') as String,
      description: (json['description'] ?? '') as String,
      status: (json['status'] ?? '') as String,
      createdAt: json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
    );
  }
}
